src/lib/components/ChatInput.svelte

```diff
@@ -1,76 +1,302 @@
+  import { toast } from "svelte-french-toast";
+  import { untrack } from "svelte";
 
   type Props = {
-    content: Record<any, any>;
+    content: Record<string, unknown>;
     users: Item[];
     context: Item[];
     onEnter: () => void;
   };
 
   let { content = $bindable({}), users, context, onEnter }: Props = $props();
   let element: HTMLDivElement | undefined = $state();
-  let extensions = $derived([
-    StarterKit.configure({ heading: false }),
-    Placeholder.configure({ placeholder: "Write something ..." }),
-    initKeyboardShortcutHandler({ onEnter }),
-    initUserMention({ users }),
-    initSpaceContextMention({ context }),
-  ]);
-
   let tiptap: Editor | undefined = $state();
+  let fileInput: HTMLInputElement | null = $state(null);
 
-  onMount(() => {
-    tiptap = new Editor({
-      element,
-      extensions,
-      content: content.type ? content : { type: "doc", children: [] },
-      editorProps: {
-        attributes: {
-          class:
-            "w-full px-3 py-2 rounded bg-base-300 text-base-content outline-none",
-        },
-      },
-      onUpdate: (ctx) => {
-        content = ctx.editor.getJSON();
-      },
-    });
+  // Flag to prevent circular updates between editor and content
+  let isInternalUpdate = $state(false);
+
+  // Track dependencies to avoid unnecessary editor recreation
+  let lastDeps = $state({
+    users: JSON.stringify(users),
+    context: JSON.stringify(context),
+    onEnter: onEnter.toString()
   });
 
+  // Action functions for event handling
+  export function handleClick(node: HTMLElement) {
+    const clickHandler = () => {
+      fileInput?.click();
+    };
+
+    node.addEventListener('click', clickHandler);
+
+    return {
+      destroy() {
+        node.removeEventListener('click', clickHandler);
+      }
+    };
+  }
+
+  export function handleChange(node: HTMLElement) {
+    const changeHandler = (event: Event) => {
+      handleFileProcess(event);
+    };
+
+    node.addEventListener('change', changeHandler);
+
+    return {
+      destroy() {
+        node.removeEventListener('change', changeHandler);
+      }
+    };
+  }
+
+  export function handlePaste(node: HTMLElement) {
+    const pasteHandler = async (event: ClipboardEvent) => {
+      // Get clipboard data
+      const items = event.clipboardData?.items;
+      if (!items) return;
+
+      // Check for image data in clipboard
+      for (const item of items) {
+        if (item.type.startsWith('image/')) {
+          // Get the image file from clipboard
+          const file = item.getAsFile();
+          if (!file) continue;
+
+          // Prevent default paste behavior
+          event.preventDefault();
+
+          // Show loading indicator
+          const uploadButton = document.querySelector('[aria-label="Upload image"]');
+          if (uploadButton) {
+            uploadButton.setAttribute('disabled', 'true');
+            uploadButton.setAttribute('title', 'Uploading...');
+          }
```

Comment:
It would be great if we could actually show a loading indicator. This just darkens the button a bit perhaps a subtle spinner inside the chat input while we load.